kornia/augmentation/container/utils.py

```diff
@@ -299,7 +311,17 @@ def inverse(
         elif isinstance(module, kornia.augmentation.ImageSequential):
             temp = module.apply_inverse_func
             module.apply_inverse_func = MaskApplyInverse
-            input = module.inverse(input, params=None if param is None else cast(List, param.data))
+            if isinstance(module, kornia.augmentation.AugmentationSequential):
+                input = cast(
```

Comment:
Instead of cast, handle the case
```suggestion
                _out = module.inverse(...)
                if isinstance(_out, list):
                    input = stack(_out)
                else:
                    input = _out
```

Reply:
@johnnv1 what do you think about targeting something like
```python
input = KORNIA_UNWRAP(_out)
```
with some smart internal to try to autocast things to tensors ?
i've been playing with a similar idea here: https://github.com/kornia/kornia/pull/1987/files#diff-0b816ce6346cf40d134afe07938df4d0685904242b27b87eae1d0ef091436913R25

Reply:
This can help a lot, and also remove plenty of casts from the codebase without adding too much code

Reply:
great -- i'll try to land that PR soon and we can evolve that piece of code to a more concise api

Reply:
Should I leave it like this for now and do the changes on all locations when the new solution is available then?

Reply:
I don't know. What does @edgarriba prefer?

Reply:
let update once is available so that we can keep moving